# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush.glsl
# Commit: 0259eb7967f4
# Full Hash: 0259eb7967f41e2476320a899efa85e59a3fe837
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2019-05-01 15:57:23
# Regressor Bug: 1540200
# File Overlap Count: 1
# Description:
#   Bug 1540200 - Part 3. Reuse snapping offsets calculated on the CPU in the WebRender shaders. r=kvark
#   
#   Historically we calculated the snapping offsets in the GPU shaders.
#   Because this information is always needed on the CPU side, we now just
#   pass the values into the shader instead of recalculating again. This
# ==============================================================================

diff -r 67e8e1f2f00c -r 0259eb7967f4 gfx/wr/webrender/res/brush.glsl
--- a/gfx/wr/webrender/res/brush.glsl	Wed Apr 10 14:20:34 2019 -0400
+++ b/gfx/wr/webrender/res/brush.glsl	Fri Apr 12 09:34:25 2019 -0400
@@ -24,7 +24,6 @@
 #define BRUSH_FLAG_SEGMENT_REPEAT_X             4
 #define BRUSH_FLAG_SEGMENT_REPEAT_Y             8
 #define BRUSH_FLAG_TEXEL_RECT                  16
-#define BRUSH_FLAG_SNAP_TO_PRIMITIVE           32
 
 #define INVALID_SEGMENT_INDEX                   0xffff
 
@@ -65,7 +64,6 @@
 
     // Write the normal vertex information out.
     if (transform.is_axis_aligned) {
-        bool snap_to_primitive = (brush_flags & BRUSH_FLAG_SNAP_TO_PRIMITIVE) != 0;
         vi = write_vertex(
             segment_rect,
             ph.local_clip_rect,
@@ -73,7 +71,7 @@
             transform,
             pic_task,
             ph.local_rect,
-            snap_to_primitive
+            ph.snap_offsets
         );
 
         // TODO(gw): transform bounds may be referenced by